---
title: "deseq2 visualiztion"
output: html_document
date: "2023-11-28"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### load packages

```{r run once}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install(version = "3.12")
biocLite("DESeq2")
library(DESeq2)
```

## DESeq2

### load in count data 

```{r}

options(scipen = 999) #disables printing results in scientific notation

data <- read.delim("../output/kallisto.isoform.counts.matrix") 

rownames(data) <- data$X

data <- data[,-1]

#ensures all data as integer
data <- round(data, digits = 0)

head(data)
```

### build objects; specify which columns are in groups

```{r}
deseq2.colData <- data.frame(condition=factor(c(rep("Control", 2), rep("Treatment", 2))), 
                             type=factor(rep("single-read", 4)))
rownames(deseq2.colData) <- colnames(data)
deseq2.dds <- DESeqDataSetFromMatrix(countData = data,
                                     colData = deseq2.colData, 
                                     design = ~ condition)

```

### run analysis

```{r}

deseq2.dds <- DESeq(deseq2.dds)
deseq2.res <- results(deseq2.dds)
#deseq2.res <- deseq2.res[order(rownames(deseq2.res)) , ]

head(deseq2.res)
```

### Count number of hits with adjusted p-value less then 0.05

```{r}
DEG <- deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ]

tmp <- deseq2.res

write.table(DEG,"analyses/0309-DEG.tab", row.names = T)
```

## Visualization 

### Volcano

```{r volcano}
plot(tmp$baseMean, tmp$log2FoldChange, pch=20, cex=0.45, ylim=c(-10, 10), log="x", col="darkgray",
     main="DEG Estradiol versus Control  (pval <= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")

# Getting the significant points and plotting them again so they're a different color
tmp.sig <- deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ]
points(tmp.sig$baseMean, tmp.sig$log2FoldChange, pch=20, cex=0.45, col="red")

# 2 FC lines
abline(h=c(-1,1), col="blue")
```

### Heatmap

```{r}
#load libraries for heatmap
library("RColorBrewer")
library( "genefilter" )
library("pheatmap")
```

```{r}
#Regularized log transformation
rld <- rlog( deseq2.dds )

#Get 25 top varying genes
topVarGenes <- head( order( rowVars( assay(rld) ), decreasing=TRUE ), 25)
```

```{r}
#make heatmap of top 25 differentially expressed genes
top25Counts<-assay(rld)[topVarGenes,]

pheatmap(top25Counts,scale = "row",show_rownames = TRUE, treeheight_col = 1, fontsize_col = 12, fontsize_row = 5)
```

### Sig. DEG Heatmap

```{r}
# make a heatmap using only our significant DEGs

#import data
DEGCounts <- read.table("analyses/0309-DEGCounts.tab", header=TRUE, quote="\"")

#select subset of just gene and sample columns
DEGHeatmap <- DEGCounts[-c(2:7)]

#establish that the first column is row names 
rownames(DEGHeatmap) <- DEGHeatmap$gene

#get rid of the redundant row names column that will be created.
DEGHeatmap <- DEGHeatmap[,-1]

#make heatmap
pheatmap(DEGHeatmap, scale = "row",show_rownames = TRUE, treeheight_col = 1, fontsize_col = 10, fontsize_row = 10)
```
