---
title: "deseq2 visualiztion"
output: html_document
date: "2023-11-28"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### load packages

```{r run once}
### only run if packages not already installed

# if (!requireNamespace("BiocManager", quietly = TRUE))
#  install.packages("BiocManager")
# BiocManager::install("biocLite")
library(DESeq2)
library(tidyverse)
```

## DESeq2

### load in count data

```{r}

options(scipen = 999) #disables printing results in scientific notation

data <- read_delim("../output/kallisto.isoform.counts.matrix") 

## here we want to specify our row names from out first column of data. it can be a little funky, but saving this order should work:

# create object for row names from the first column
# used the $ index because brackets were giving trouble
rownames <- data$...1
head(rownames)

# remove that first column now that it's values are stored
data <- data[,-1]
head(data)

#ensures all data as integer
data <- round(data, digits = 0)
head(data)

# assign the row names to the data using the object created 
rownames(data) <- rownames
head(data)

```

### build objects; specify which columns are in groups

```{r}
deseq2.colData <- data.frame(condition = factor(c(rep("Control", 15), rep("Treatment", 15))), type = factor(rep("paired-read", 30)))

rownames(deseq2.colData) <- colnames(data)

deseq2.dds <- DESeqDataSetFromMatrix(countData = data,
                                     colData = deseq2.colData, 
                                     design = ~ condition)

```

### run analysis

```{r}

deseq2.dds <- DESeq(deseq2.dds)
deseq2.res <- results(deseq2.dds)
deseq2.res <- deseq2.res[order(rownames(deseq2.res)),]

summary(deseq2.res)
```

### Count number of hits with adjusted p-value less then 0.05

```{r}
DEG <- deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ]
dim(DEG)

tmp <- deseq2.res

write.table(DEG,"../output/1213-DEG.tab", row.names = T)
```

## Visualization

### Volcano

```{r volcano}
plot(tmp$baseMean, tmp$log2FoldChange, pch=20, cex=0.45, ylim=c(-10, 10), log="x", col="darkgray",
     main="Treatmetn versus Control  (pval <= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")

# Getting the significant points and plotting them again so they're a different color
tmp.sig <- deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ]
points(tmp.sig$baseMean, tmp.sig$log2FoldChange, pch=20, cex=0.45, col="red")

# 2 FC lines
abline(h=c(-1,1), col="blue")
```

### Heatmap

```{r}
#load libraries for heatmap
library("RColorBrewer")
# library( "genefilter" )
library("pheatmap")
```

```{r}
#Regularized log transformation
rld <- rlog( deseq2.dds, blind = FALSE )

#Get 25 top varying genes
topVarGenes <- head( order( rowVars (x = assay(rld), useNames = FALSE ), decreasing=TRUE ), 25)
```

```{r}
# make heatmap of top 25 differentially expressed genes
# get top 25 counts
top25Counts<-assay(rld)[topVarGenes,]
head(top25Counts)

# make the heatmap. Cluster_cols = FALSE neccessary to get the M-T and M-C columns next to eachother (defult leaves em randomized)
pheatmap(top25Counts,scale = "row",show_rownames = TRUE, treeheight_col = 1, fontsize_col = 12, fontsize_row = 5, cluster_cols = FALSE)

```

```{r}
# log2_changes <- deseq2.res[, "log2FoldChange"]

# Filter for significant genes (adjusted p-value < 0.05)
significant_results <- subset(deseq2.res, padj < 0.05)

# Sort results by absolute log2 fold change
significant_results <- significant_results[order(abs(significant_results$log2FoldChange), decreasing = TRUE), ]

# Get top 25 significant genes
top_25_genes <- head(significant_results, 25)

# Extract log2 fold change values for top 25 genes
top_25_genes_ids <- rownames(top_25_genes)
top_25_genes_log2fc <- top_25_genes$log2FoldChange

# Create a matrix containing log2 fold change values of top 25 genes
top_25_genes_matrix <- count_matrix[top_25_genes_ids, ]

# Plot heatmap for log2 fold change of top 25 genes
pheatmap(top_25_genes_matrix, 
         cluster_rows = TRUE, 
         cluster_cols = FALSE,
         main = "Heatmap of Log2 Fold Change for Top 25 DEGs",
         #color = colorRampPalette(c("blue", "yellow", "red"))(100),  # Choose a color palette
         breaks = seq(-4, 4, length.out = 101),  # Adjust the breaks for better color resolution
         fontsize = 8)  # Adjust fontsize as needed




```
